File: Server/src/main/java/com/daffodil/renters/core/model/beans/Property.java
package com.daffodil.renters.core.model.beans;

import com.daffodil.renters.core.model.beans.postables.Amenities;
import com.daffodil.renters.core.model.beans.postables.ParkingSpot;
import com.daffodil.renters.core.model.beans.postables.Seller;

import java.util.Date;
import java.util.List;

public class Property extends PropertySkeletal {

    String description;
    Amenities amenities;
    Seller seller;
    long securityDeposit;
    long brokerage;
    int ageOfProperty;
    int lockInPeriod;
    Date listedOn;
    List<ParkingSpot> parkingSpots;

}

File: Android-Client/app/src/main/java/com/daffodil/renters/model/postables/Property.kt
package com.daffodil.renters.model.postables

import java.util.Date

data class Property(
    var id: Long?,
    var unit: String?,
    var description: String?,
    var propertyType: PROPERTY_TYPE? = PROPERTY_TYPE.APARTMENT,
    var furnishingType: FURNISHING_TYPE? = FURNISHING_TYPE.UNFURNISHED,
    var area: Int?,
    var rent: Long?,
    var roommates: Boolean?,
    var securityDeposit: Long?,
    var brokerage: Long?,
    var lockInPeriod: Int?,
    var listedOn: Date?,
    var building: Building?,
    var seller: Seller?,
    var amenities: Amenities?,
    var rooms: List<Room>?,
    var parkingSpots: List<ParkingSpot>?
) {


    enum class PROPERTY_TYPE {
        APARTMENT,
        INDEPENDENT_FLOOR,
        INDEPENDENT_HOUSE,
        PG
    }

    enum class FURNISHING_TYPE {
        FULLY_FURNISHED,
        SEMI_FURNISHED,
        UNFURNISHED
    }
}
